### ISSUE TITLE
[DOC] Advanced transfer function requirements document

### ISSUE THREAD

**Issue by ylic** [KuChainNetwork/Project-Decalogue#12]
**Type of Proposal**

- [x] Feature Request (e.g. functionality) 

- [x] Economic Model

- [x] Underlying Technology (e.g. performance) 

- [x] Application Development

- [ ] Others

  

**Background**

In some special scenarios, we want the transfer to take effect not immediately but over a period of time. In other transaction scenarios, we want a third party to witness the transfer.

**Describe The Proposal (No more than 50 words)** 

Add the function of delay transfer and guarantee transfer.

**Business Model (Optional for underlying technology)** 

Alice signs a delayed transaction, pays Bob's address, and sets the transaction nLocktime to the next 3 months. Alice sends Bob the deal and puts it on hold. With this deal, Alice and Bob know:    

Bob cannot complete the transaction for conversion until three months have passed.

Bob can accept the transaction after 3 months.

Alice and Bob trade a commodity. When they meet for the first time, they don't know each other, but they are friends with Tom. Therefore, When Alice transfers money to Bob, she asks Tom to participate in the transfer and witness the transfer process. So Alice initiates a secured transaction with Bob, referring to Tom as the guarantor. This transfer can only reach Bob's account after Tom's confirmation.

**Expected Outcome**
1. For delayed transaction, time lock parameter can be added on the basis of ordinary transaction. Before the transaction reaches the time lock, the payee cannot use the transferred amount, and the payer cannot withdraw the transferred amount.

2. For secured transactions, a witness can be added on the basis of ordinary transactions, and the transaction will take effect only after it is confirmed by the witness. If the witness fails to confirm within the specified time, the transaction fails.

**Timetable (Optional)** 

Estimate the timetable of your plan. 

| Implementation Steps | Time Consuming |
| -------------------- | -------------- |
| step 1               | modify asset module |
| step 2               | testing           |
| Total duration       | one month  |

**Contact Information**

ylic@263.net
